trinity/rpc/modules/net.py

```diff
@@ -10,3 +10,21 @@ def version(self):
         Returns the current network ID.
         """
         return str(self._chain.network_id)
+
+    def peerCount(self):
+        """
+        Return the number of peers that are currently connected to the node
+        """
+        if self._p2p_server.peer_pool is None:
+            return '0x0'
```

Comment:
I'm curious about this logic and why it's here.  Seems like the `net_` API should not be runnable if there is no peer pool.

Reply:
Well, parts of the `net` module are independent from the peer pool (e.g. version) and by the time that the peer pool module is created, we don't have a peer pool yet. The peer pool is created asynchronously and it takes several seconds until it is set here: https://github.com/ethereum/py-evm/blob/f66d296074c0603eefa4e4c718421956d70d183e/p2p/server.py#L262

In other words, when you start trinity, the `net` module is already set up and answers to `net_peerCount` request and does therefore return `0x0` for when the `peer_pool` is still `None`. Which at least technically is correct :)

Reply:
Hrm, can you open an issue to track this as I think that we should fix that.

Reply:
Maybe if `peer_pool is None` should return a JSON-RPC error indicating that it's not ready?

Reply:
> in fact our p2p subsystem is still bootstrapping and we have in fact 0 connected peers at that time.

👍

Reply:
Wouldn't `hex(0)` cut it if you dislike the hardcoded string?

Reply:
@pipermerriam I created #877 regarding the `PeerPool` issue.